template/cordova/lib/prepare.js

```diff
@@ -425,19 +451,39 @@ module.exports.prepare = function (cordovaProject) {
     AppxManifest.purgeCache();
 
     // Update own www dir with project's www assets and plugins' assets and js-files
-    return Q.when(updateWwwFrom(cordovaProject, this.locations))
+    return Q.when(updateWww(cordovaProject, this.locations))
     .then(function () {
         // update project according to config.xml changes.
         return updateProjectAccordingTo(self._config, self.locations);
     })
     .then(function () {
-        copyImages(cordovaProject.projectConfig, self.root);
+        copyImages(cordovaProject, self.locations);
     })
     .then(function () {
         events.emit('verbose', 'Prepared windows project successfully');
     });
 };
 
+module.exports.clean = function (options) {
+    // A cordovaProject isn't passed into the clean() function, because it might have
+    // been called from the platform shell script rather than the CLI. Check for the
+    // noPrepare option passed in by the non-CLI clean script. If that's present, or if
+    // there's no config.xml found at the project root, then don't clean prepared files.
+    var projectRoot = path.resolve(this.root, '../..');
+    var projectConfigFile = path.join(projectRoot, 'config.xml');
```

Comment:
That's a reasonable question. I didn't want the clean function to depend on a previous prepare having completed successfully (so that the platform config.xml was merged). But I'm not sure that's best, and maybe it would be cleaner just to use that file.

I did this the same way for the other 2 platforms, so this discussion would apply there also.